storage/replicate_queue.go

```diff
@@ -83,12 +90,21 @@ func (rq replicateQueue) shouldQueue(now proto.Timestamp, repl *Replica) (should
 }
 
 func (rq replicateQueue) process(now proto.Timestamp, repl *Replica) error {
-	zone, err := lookupZoneConfig(rq.gossip, repl)
+	// Load the system config.
+	cfg, err := rq.gossip.GetSystemConfig()
 	if err != nil {
 		return err
 	}
 
+	// TODO(marc): shouldn't we be checking whether the range needs to be split?
```

Comment:
Yeah, I think that `splitQueue` is probably the only one that should operate on ranges that should be split, and all the others should check for potential splits both in `shouldQueue` and `process`. Would be nice to move this into the scanner or base queue but I'm not sure if we can do it cleanly. (maybe a new method like `needsLeaderLease`?)